features/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/recorder/base64/Base64Serializer.kt

```diff
@@ -93,17 +95,51 @@ internal class Base64Serializer private constructor(
         imageWireframe: MobileSegment.Wireframe.ImageWireframe,
         base64SerializerCallback: Base64SerializerCallback?
     ) {
-        val base64String = convertBmpToBase64(drawable, bitmap, shouldCacheBitmap)
-        finalizeRecordedDataItem(base64String, imageWireframe, base64SerializerCallback)
+        val byteArray = compressBitmap(bitmap)
+        val base64String = convertBmpToBase64(byteArray, drawable, bitmap, shouldCacheBitmap)
+        val hash = calculateHash(drawable, byteArray)
+        finalizeRecordedDataItem(base64String, hash, imageWireframe, base64SerializerCallback)
+    }
+
+    private fun calculateHash(
+        drawable: Drawable,
+        byteArray: ByteArray
+    ): String {
+        val md5Generator = MD5HashGenerator(logger)
+        val hashValue = md5Generator.generate(byteArray)
+
+        if (hashValue.isNotEmpty()) {
```

Comment:
having hash equal to empty string is something strange, it is better to make `generate` method return `null` in case of an error.